vterm.el

```diff
@@ -1103,7 +1106,7 @@ More information see `vterm--prompt-tracking-enabled-p' and
 
 (defun vterm--at-prompt-p ()
   "Return t if the cursor postion is at shell prompt."
-  (= (point) (vterm--get-prompt-point)))
+  (= (point) (or (vterm--get-prompt-point) 0)))
```

Comment:
I think the name is misleading (ie, it doesn't convey what the function does): `vterm--at-prompt-p` should return t only when the point is at the prompt, if `vterm--get-prompt-point` returns nil, it means that we cannot locate the prompt, so this function should fail.

Also, we should document that if the prompt detection fails, we use the beginning of the line instead.

Reply:
when tail -f  
and all promp  disappeard  it return nil

Reply:
`(or nil 0)` returns 0, so I don't see how this function can return nil

Reply:
when `(vterm--get-prompt-point)` is a number , `(or  (vterm--get-prompt-point) 0) `would return that number.
when  `(vterm--get-prompt-point)` is nil   `(or  (vterm--get-prompt-point) 0)`  would return 0 

`(= (point) 0)` would return nil, because `(point)` always >0

Reply:
Gotcha! I completely overlooked the function. Thanks for the detailed explanation.

Then, it looks good to me.